frame/nomination-pools/benchmarking/src/lib.rs

```diff
@@ -0,0 +1,610 @@
+
+		// Some last checks that storage items we expect to get cleaned up are present
+		assert!(pallet_staking::Ledger::<T>::contains_key(&pool_account));
+		assert!(BondedPools::<T>::contains_key(&1));
+		assert!(SubPoolsStorage::<T>::contains_key(&1));
+		assert!(RewardPools::<T>::contains_key(&1));
+		assert!(Delegators::<T>::contains_key(&depositor));
+		assert!(frame_system::Account::<T>::contains_key(&reward_account));
+
+		whitelist_account!(depositor);
+	}: withdraw_unbonded_other(Origin::Signed(depositor.clone()), depositor.clone(), s)
+	verify {
+		// Pool removal worked
+		assert!(!pallet_staking::Ledger::<T>::contains_key(&pool_account));
+		assert!(!BondedPools::<T>::contains_key(&1));
+		assert!(!SubPoolsStorage::<T>::contains_key(&1));
+		assert!(!RewardPools::<T>::contains_key(&1));
+		assert!(!Delegators::<T>::contains_key(&depositor));
+		assert!(!frame_system::Account::<T>::contains_key(&pool_account));
+		assert!(!frame_system::Account::<T>::contains_key(&reward_account));
+
+		// Funds where transferred back correctly
+		assert_eq!(
+			CurrencyOf::<T>::free_balance(&depositor),
+			// gets bond back + rewards collecting when unbonding
+			min_create_bond * 2u32.into()
+		);
+	}
+
+	create {
+		let min_create_bond = MinCreateBond::<T>::get()
+			.max(T::StakingInterface::minimum_bond())
+			.max(CurrencyOf::<T>::minimum_balance());
+		let depositor: T::AccountId = account("depositor", USER_SEED, 0);
+
+		// Give the depositor some balance to bond
+		CurrencyOf::<T>::make_free_balance_be(&depositor, min_create_bond * 2u32.into());
+
+		// Make sure no pools exist as a pre-condition for our verify checks
+		assert_eq!(RewardPools::<T>::count(), 0);
+		assert_eq!(BondedPools::<T>::count(), 0);
+
+		whitelist_account!(depositor);
+	}: _(
+			Origin::Signed(depositor.clone()),
+			min_create_bond,
+			depositor.clone(),
+			depositor.clone(),
+			depositor.clone()
+		)
+	verify {
+		assert_eq!(RewardPools::<T>::count(), 1);
+		assert_eq!(BondedPools::<T>::count(), 1);
+		let (_, new_pool) = BondedPools::<T>::iter().next().unwrap();
+		assert_eq!(
+			new_pool,
+			BondedPoolInner {
+				points: min_create_bond,
+				state: PoolState::Open,
+				delegator_counter: 1,
+				roles: PoolRoles {
+					depositor: depositor.clone(),
+					root: depositor.clone(),
+					nominator: depositor.clone(),
+					state_toggler: depositor.clone(),
+				},
+			}
+		);
+		assert_eq!(
+			T::StakingInterface::active_stake(&Pools::<T>::create_bonded_account(1)),
+			Some(min_create_bond)
+		);
+	}
+
+	nominate {
+		// Create a pool
+		let min_create_bond = MinCreateBond::<T>::get()
+			.max(T::StakingInterface::minimum_bond())
+			.max(CurrencyOf::<T>::minimum_balance());
+		let (depositor, pool_account) = create_pool_account::<T>(0, min_create_bond);
+
+		// Create some accounts to nominate. For the sake of benchmarking they don't need to be
+		// actual validators
+		 let validators: Vec<_> = (0..T::MaxNominations::get())
```

Comment:
Not sure if its worth, but changing this one and the `set_metadata` to have a length parameter instead of using max length would give us more precise weights.

Reply:
dynamic nomination weight: 9190415a450c7f8b78b763d235bc7cc55608ee85

Reply:
dynamic set metadata weight: d31afbfe181177196599c43429599f99371556af